### PR TITLE
zap_replace_klog

### PR DESCRIPTION
**Pull request by jyyds**
**What this PR does / why we need it**:

**Will this PR make the community happier**? 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**How is this PR tested**
- [ ] unit test
- [ ] e2e test
- [x] other (please specify)

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Review by tomqin93 — CHANGES_REQUESTED** [review]
@jyyds please sync your branch with main to resolve conflicts

File: pkg/deviceshifu/deviceshifuhttp/deviceshifuhttp.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func New(deviceShifuMetadata *deviceshifubase.DeviceShifuMetaData) (*DeviceShifu
 				mux.HandleFunc("/"+instruction, handler.commandHandleFunc())
 			}
 		default:
-			klog.Errorf("EdgeDevice protocol %v not supported in deviceShifu_http_http", protocol)
+			zlog.Errorf("EdgeDevice protocol %v not supported in deviceShifu_http_http", protocol)
```
Comment (tomqin93) [inline pkg/deviceshifu/deviceshifuhttp/deviceshifuhttp.go:99]: Let's use the `logger` name directly, no need to rename it to zlog, same for other occurrences

**Review by tomqin93 — COMMENTED** [review]
@jyyds added 1 comment, please take a look

File: pkg/deviceshifu/deviceshifuhttp/deviceshifuhttp.go (hunk 1 above)
Comment (jyyds) [reply pkg/deviceshifu/deviceshifuhttp/deviceshifuhttp.go]: ok

**Review by tomqin93 — APPROVED** [review]
LGTM

**Review by tomqin93 — CHANGES_REQUESTED** [review]
@jyyds please fix build issue:
```
 > [builder 10/10] RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go:
#17 1.076 pkg/k8s/crd/usermetrics/utils/utils.go:14:2: no required module provides package github.com/edgenesis/shifu/pkg/logger; to add it:
#17 1.076 	go get github.com/edgenesis/shifu/pkg/logger
------
ERROR: failed to solve: executor failed running [/bin/sh -c CGO_ENABLED=0 GOOS=$TARGETOS GOARCH=$TARGETARCH go build -a -o manager main.go]: exit code: 1
##[error]Bash exited with code '1'.

```

File: pkg/k8s/crd/Dockerfile [hunk 2]
```diff
@@ -17,6 +17,9 @@ COPY pkg/k8s/crd/main.go main.go
 COPY pkg/k8s/crd/usermetrics pkg/k8s/crd/usermetrics
 COPY pkg/k8s/api pkg/k8s/api
 COPY pkg/k8s/controllers pkg/k8s/controllers
+COPY pkg/logger pkg/logger
+
+
```
Comment (tomqin93) [inline pkg/k8s/crd/Dockerfile:22]: remove extra blank lines

**Review by tomqin93 — CHANGES_REQUESTED** [review]
@jyyds addedd 1 comment, please take a look

File: pkg/k8s/crd/Dockerfile (hunk 2 above)
Comment (jyyds) [reply pkg/k8s/crd/Dockerfile:22]: ok

File: pkg/deviceshifu/deviceshifuplc4x/deviceshifuplc4x.go [hunk 3]
```diff
@@ -114,17 +114,22 @@ func (ds *DeviceShifu) writeCommandHandlerPlc4x() http.HandlerFunc {
...
-		ctx, cancel := context.WithTimeout(context.Background(), time.Second*timeout)
-		defer cancel()
-		resultData := <-plc4xRequest.ExecuteWithContext(ctx)
+		var resultData model.PlcWriteRequestResult
+		select {
+		case resultData = <-plc4xRequest.Execute():
+		case <-time.After(time.Second * timeout):
+			logger.Errorf("Timeout when send request to device")
+			http.Error(w, "Timeout when send request to device", http.StatusInternalServerError)
+			return
```
Comment (tomqin93) [inline pkg/deviceshifu/deviceshifuplc4x/deviceshifuplc4x.go:128]: let's sync with main to avoid previous commits

**Review by tomqin93 — CHANGES_REQUESTED** [review]
@jyyds added 1 comment

File: pkg/deviceshifu/deviceshifuplc4x/deviceshifuplc4x.go (hunk 3 above)
Comment (jyyds) [reply pkg/deviceshifu/deviceshifuplc4x/deviceshifuplc4x.go:128]: ok

**Pull request closed without merging**